task7/task7.js

```diff
@@ -0,0 +1,424 @@
+       createdAt: new Date('2020-03-17T23:20:00'),
+       author: 'пщфышра',
+       hashTags: ['#hashtag1', '#hashtag2', '#sport', '#PS4'],
+       likes: ['Виктор Комаров', 'Antony Pavlov', 'Gorrive Curipita']
+      },
+
+      {
+       id: '18',
+       description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+       createdAt: new Date('2020-03-17T13:00:20'),
+       author: 'фтоф фдла',
+       hashTags: ['#hashtag1', '#hashtag2', '#sport', '#EA', '#PS4'],
+       likes: ['Виктор Комаров', 'Antony Pavlov', 'Gorrive Curipita']
+      },
+
+      {
+       id: '19',
+       description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+       createdAt: new Date('2000-03-17T23:06:00'),
+       author: 'фафф ф ф ф',
+       hashTags: ['#hashtag1', '#hashtag2', '#sport', '#EA', '#PS4'],
+       likes: ['Виктор Комаров', 'Antony Pavlov', 'Gorrive Curipita']
+      },
+
+      {
+       id: '20',
+       description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+       createdAt: new Date('2020-05-17T23:00:00'),
+       author: 'Goro',
+       hashTags: ['#hashtag1', '#hashtag2', '#sport', '#EA', '#PS4'],
+       likes: ['Виктор Комаров', 'Antony Pavlov', 'Gorrive Curipita']
+      }
+      ];
+
+let testPost = {
+   description: 'HI ',
+   createdAt: new Date('1990-03-18T23:40:07'),
+   author: 'name surname',
+   photoLink: 'https://www.pressball.by/images/stories/2020/03/20200310231542.jpg',
+   hashTags: ['#hashtag2', '#sport'],
+   likes: ['Виктор Комаров', 'Antony Pavlov', 'Gorrive Curipita']
+};
+
+class PostList {
+  _postCollection = [];
+
+
+
+  constructor(postCollection) {
+    this._postcollection = postCollection;
+  }
+
+  static _validatePost(post) {
+    return  ((Array.isArray(post.likes))
+          &&(typeof post.id === 'string')
+          &&(typeof post.description === 'string')
+          &&(post.createdAt instanceof Date)
+          &&(typeof post.author === 'string'));
+  }
+
+  getPost(id) {     
+    return this._postcollection.find((post) => post.id === id);
+  }
+
+  addPost(post) {
+    if (typeof post != 'object') {
+      return false;
+    }
+    let temp = '1';
+    while (this._postcollection.findIndex((post) => post.id === temp) !== -1) {
+      temp = Math.random().toString(32);
+    }
+    post.id = temp;
+
+    if (!PostList._validatePost(post)) {
+      return false;
+    }
+    this._postcollection.push(post);
+    //console.log("Post added with ID: " + post.id);
+    return true;
+  }
+
+  editPost(id, post) {
+    var postIndex = this._postcollection.findIndex((post) => post.id === id);
```

Comment:
Use 'let' instead of 'var' unless you have a strong reason to use 'var' keyword.
Because using 'var', it may produce unexpected output. While using 'let', you don't need to mess with the scopes. A variable defined using 'let' is known only in the block it is defined in.
It won't lead to unexpected outputs.